### PR TITLE
Gave names to anonymous functions in the docs. Tried to be as semantic as possible.

### PR DESCRIPTION
**Pull request by catcarbonell**
Responding to issue #17200

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| buildDuration | 13s | 13.6s | ⚠️ +549ms |
| nodeModulesSize | 63.1 MB | 63.1 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.493 | 2.514 | ⚠️ +0.02 |
| / avg req/sec | 1002.7 | 994.24 | ⚠️ -8.46 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.473 | 1.507 | ⚠️ +0.03 |
| /error-in-render avg req/sec | 1696.75 | 1658.44 | ⚠️ -38.31 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| 677f882d2ed8..9339.js gzip | 10.9 kB | 10.9 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-d493e7d..42f5.js gzip | 7.17 kB | 7.17 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.8 kB | 57.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.77 kB | 6.77 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-f8905d4..dule.js gzip | 6.24 kB | 6.24 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-d2344ce..8b36.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.71 kB | 7.71 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-f8c0daf..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| index.html gzip | 1 kB | 1 kB | ✓ |
| link.html gzip | 1.01 kB | 1.01 kB | ✓ |
| withRouter.html gzip | 996 B | 996 B | ✓ |
| Overall change | 3.01 kB | 3.01 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| buildDuration | 15.3s | 15.4s | ⚠️ +76ms |
| nodeModulesSize | 63.1 MB | 63.1 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| 677f882d2ed8..9339.js gzip | 10.9 kB | 10.9 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-d493e7d..42f5.js gzip | 7.17 kB | 7.17 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.8 kB | 57.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.77 kB | 6.77 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-f8905d4..dule.js gzip | 6.24 kB | 6.24 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-d2344ce..8b36.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.71 kB | 7.71 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-f8c0daf..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _error.js | 1.05 MB | 1.05 MB | ✓ |
| 404.html | 4.34 kB | 4.34 kB | ✓ |
| hooks.html | 3.92 kB | 3.92 kB | ✓ |
| index.js | 1.05 MB | 1.05 MB | ✓ |
| link.js | 1.1 MB | 1.1 MB | ✓ |
| routerDirect.js | 1.09 MB | 1.09 MB | ✓ |
| withRouter.js | 1.09 MB | 1.09 MB | ✓ |
| Overall change | 5.4 MB | 5.4 MB | ✓ |

</details>

</details>
Commit: a25a24ab2a92b5c07fa498cc98bb4b2145a2d4b8

File: docs/api-routes/dynamic-api-routes.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ API routes support [dynamic routes](/docs/routing/dynamic-routes.md), and follow
 For example, the API route `pages/api/post/[pid].js` has the following code:
 
 ```js
-export default (req, res) => {
+export default function PostId(req, res){
```
Comment (lfades) [inline docs/api-routes/dynamic-api-routes.md:19]: Let's either use `handler` for all the names, or something that focuses on what the function returns, e.g `getPost` (and doesn't start with an uppercase)

Comment (catcarbonell) [reply docs/api-routes/dynamic-api-routes.md:19]: Sounds good. I'll go ahead and fix that. Thanks!

**Comment by awareness481** [comment]
I'm confused, why is there a duplicate PR? There's an earlier [PR](https://github.com/vercel/next.js/pull/17510) that addresses the issue already.

**Comment by catcarbonell** [comment]
> I'm confused, why is there a duplicate PR? There's an earlier [PR](https://github.com/vercel/next.js/pull/17510) that addresses the issue already.

I'm competitive. lol.

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| buildDuration | 10.8s | 11.1s | ⚠️ +332ms |
| nodeModulesSize | 63.1 MB | 63.1 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.03 | 2.051 | ⚠️ +0.02 |
| / avg req/sec | 1231.31 | 1218.76 | ⚠️ -12.55 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.167 | 1.151 | -0.02 |
| /error-in-render avg req/sec | 2142.23 | 2172.63 | +30.4 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| 677f882d2ed8..9339.js gzip | 10.9 kB | 10.9 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-d493e7d..42f5.js gzip | 7.17 kB | 7.17 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.8 kB | 57.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.77 kB | 6.77 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-f8905d4..dule.js gzip | 6.24 kB | 6.24 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-d2344ce..8b36.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.71 kB | 7.71 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-f8c0daf..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| index.html gzip | 1 kB | 1 kB | ✓ |
| link.html gzip | 1.01 kB | 1.01 kB | ✓ |
| withRouter.html gzip | 996 B | 996 B | ✓ |
| Overall change | 3.01 kB | 3.01 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| buildDuration | 12.5s | 12.2s | -323ms |
| nodeModulesSize | 63.1 MB | 63.1 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| 677f882d2ed8..9339.js gzip | 10.9 kB | 10.9 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-d493e7d..42f5.js gzip | 7.17 kB | 7.17 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.8 kB | 57.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.77 kB | 6.77 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-f8905d4..dule.js gzip | 6.24 kB | 6.24 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-d2344ce..8b36.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.71 kB | 7.71 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-f8c0daf..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | catcarbonell/next.js canary | Change |
| - | - | - | - |
| _error.js | 1.05 MB | 1.05 MB | ✓ |
| 404.html | 4.34 kB | 4.34 kB | ✓ |
| hooks.html | 3.92 kB | 3.92 kB | ✓ |
| index.js | 1.05 MB | 1.05 MB | ✓ |
| link.js | 1.1 MB | 1.1 MB | ✓ |
| routerDirect.js | 1.09 MB | 1.09 MB | ✓ |
| withRouter.js | 1.09 MB | 1.09 MB | ✓ |
| Overall change | 5.4 MB | 5.4 MB | ✓ |

</details>

</details>
Commit: 1ea9ebe85a0a699fc1fab9fd46480f61e4b6613e

**Comment by lfades** [comment]
Closing in favor of https://github.com/vercel/next.js/pull/17510

**Pull request closed without merging**